src/figuresClasses.ts

```
@@ -1,11 +1,85 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'Triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (this.a < 0 || this.b < 0 || this.c < 0) {
+      throw new Error('Sides cannot be negative');
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('Invalid triangle');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+
+    return Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape = 'Circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    this.color = color;
+    this.radius = radius;
 
-export class Rectangle implements Figure {}
+    if (this.radius < 0) {
```

The requirement states that any length `<= 0` should throw an error. This check should be `this.radius <= 0`.